cli/cmd/root.go

```diff
@@ -99,7 +92,7 @@ var RootCmd = &cobra.Command{
 }
 
 func init() {
-	defaultNamespace = getDefaultNamespace()
+	defaultNamespace = cmd.GetDefaultNamespace(kubeconfigPath, kubeContext)
```

Comment:
Can you check if changing `kubeconfigPath`, pointing to a config with a non default ns, would get properly detected here? At this point `kubeconfigPath` is an empty string, passed by value, and I think the Cobra command below won't affect it. If so, this is an existing bug that that can be addressed in a separate PR.